src/codewhisperer/tracker/codewhispererCodeCoverageTracker.ts

```diff
@@ -8,88 +8,122 @@ import * as telemetry from '../../shared/telemetry/telemetry'
 import { getLogger } from '../../shared/logger/logger'
 import { CodeWhispererConstants } from '../models/constants'
 import globals from '../../shared/extensionGlobals'
+import { vsCodeState } from '../models/model'
+import { distance } from 'fastest-levenshtein'
+
+interface CodeWhispererToken {
+    range: vscode.Range
+    text: string
+    accepted: number
+}
+
 /**
- * This singleton class is mainly used for calculating the percentage of user modification.
- * The current calculation method is (Levenshtein edit distance / acceptedSuggestion.length).
+ * This singleton class is mainly used for calculating the code written by codeWhisperer
  */
 export class CodeWhispererCodeCoverageTracker {
-    private _acceptedTokens: string[]
-    private _totalTokens: string[]
+    private _acceptedTokens: { [key: string]: CodeWhispererToken[] }
+    private _totalTokens: { [key: string]: number }
```

Comment:
Using objects as maps can be convenient but it'll cause problems for completely arbitrary strings. For example, if a file is named 'toString' you'll see some rather odd behavior. I'd just use a `Map` as we aren't using any narrowed string types to make using an object worth it.